src/Parser/Potree2BinParser.js

```diff
@@ -0,0 +1,136 @@
+import * as THREE from 'three';
+
+// Create enums for different types of workers
+const WORKER_TYPE = {
+    DECODER_WORKER_BROTLI: 'DECODER_WORKER_BROTLI',
+    DECODER_WORKER: 'DECODER_WORKER',
+};
+
+const workers = {};
+
+function createWorker(type) {
+    if (type === WORKER_TYPE.DECODER_WORKER_BROTLI) {
+        return new Worker(
+            /* webpackChunkName: "potree2-brotli-decoder.worker" */ new URL('../Workers/potree2-brotli-decoder.worker.js', import.meta.url),
+            { type: 'module' },
+        );
+    } else if (type === WORKER_TYPE.DECODER_WORKER) {
+        return new Worker(
+            /* webpackChunkName: "potree2-decoder.worker" */ new URL('../Workers/potree2-decoder.worker.js', import.meta.url),
+            { type: 'module' },
+        );
+    } else {
+        throw new Error('Unknown worker type');
+    }
+}
+
+function getWorker(type) {
+    if (!workers[type]) {
+        workers[type] = [];
+    }
+
+    if (workers[type].length === 0) {
+        const worker = createWorker(type);
+        workers[type].push(worker);
+    }
+
+    const worker = workers[type].pop();
+    return worker;
+}
+
+function returnWorker(type, worker) {
+    workers[type].push(worker);
+}
+
+export default {
+    /** @module Potree2BinParser */
+    /** Parse .bin PotreeConverter 2.0 format and convert to a THREE.BufferGeometry
+     * @function parse
+     * @param {ArrayBuffer} buffer - the bin buffer.
+     * @param {Object} options
+     * @param {string[]} options.in.pointAttributes - the point attributes information contained in metadata.js
+     * @return {Promise} - a promise that resolves with a THREE.BufferGeometry.
+     *
+     */
+    parse: function parse(buffer, options) {
+        return new Promise((resolve) => {
+            const layer = options.out;
+            const metadata = layer.metadata;
```

Comment:
Pass `metadata` as `options.in.metadata` since this is an option from the `source` input (i.e. `metadata.json`).
```suggestion
            const metadata = options.in.metadata;
```

Reply:
Ok